ast/src/analyzed/mod.rs

```diff
@@ -322,13 +322,13 @@ impl<T> Analyzed<T> {
             .for_each(|definition| definition.post_visit_expressions_mut(f))
     }
 
-    /// Retrieves (col_name, col_idx, offset) of each public witness in the trace.
-    pub fn get_publics(&self) -> Vec<(String, usize, usize)> {
-        let mut publics = self
-            .public_declarations
+    /// Retrieves a map public_id -> (public_name, col_idx, offset) of publics.
+    pub fn get_publics(&self) -> BTreeMap<u64, (String, usize, usize)> {
+        self.public_declarations
             .values()
             .map(|public_declaration| {
-                let column_name = public_declaration.referenced_poly_name();
+                let public_id = public_declaration.id;
+                let public_name = public_declaration.referenced_poly_name();
```

Comment:
```suggestion
                let public_name = public_declaration.name;
```
The other one is the name of the col on the rhs of the public declaration